### DISCUSSION TITLE
Fastapi cannot be effectively restarted after forced shutdown (ctrl+c)

### DISCUSSION THREAD

**Discussion by Bruce337f** [fastapi/fastapi#10605] [Questions] [answered]
### Privileged issue

- [X] I'm @tiangolo or he asked me directly to create an issue here.

### Issue Content

When I use ctrl + c to forcefully close the current fastapi program, the cmd interface only stays at Finishsever and the shuttingdown code line cannot be seen. At this time, fastapi can still be used normally and is not effectively closed.

But when I directly closed the cmd program running by fastapi by killing the process, and restarted fastapi through python, it could no longer automatically and **effectively restart** as the code file changed _(I modified the code at will, but fastapi read The code file still stays at the last update before closing, and subsequent updates and changes are ignored by uvicorn_).

I can only make the fastapi program run normally by restarting the computer, but the **cost of** restarting the computer is too high. I want to keep fastapi and uvicorn closed and started normally. What should I do?

**Comment by Kludex** [comment, marked as answer]
Bump uvicorn to `0.24.0`, and try again.

**Comment by Bruce337f** [comment]
Thank you for your reply!
Didn't think it was a version issue, I will try it!

**Comment by vak0160** [comment]
Hi, I have the same issue with uvicorn `0.25.0`. I'm using Windows 10 and python `3.11.2`.

```
fastapi = "^0.108.0"
uvicorn = {extras = ["standard"], version = "^0.25.0"}
python-jose = {extras = ["cryptography"], version = "^3.3.0"}
passlib = {extras = ["bcrypt"], version = "^1.7.4"}
attrs = "^23.1.0"
python-dotenv = "^1.0.0"
environ-config = "^23.2.0"
pydantic = {extras = ["email"], version = "^2.5.3"}
cattrs = "^23.2.3"
python-multipart = "^0.0.6"
bcrypt = "4.0.1"
```
